rust/relay/src/server.rs

```diff
@@ -755,7 +758,7 @@ where
             .map_err(|_| error_response(StaleNonce, request))?;
 
         message_integrity
-            .verify(&self.auth_secret, username.name(), now)
+            .verify(self.auth_secret.expose_secret(), username.name(), now)
```

Comment:
`.verify` comes from an extension trait actually so you can modify the signature and delay unwrapping the secret even further.

Reply:
Nice, done!